### DISCUSSION TITLE
Problem when use npm run build with Astro

### DISCUSSION THREAD

**Discussion by basshamut** [primefaces/community#1751] [PrimeReact] [answered]
I have this problem when I try to build my project, I am using Astro + React + PrimeReact:

$ npm run build

> demo-astro-react-vite@0.0.1 build
> astro build

22:54:45 [build] output: "static"
22:54:45 [build] directory: C:\dev\Demos\javascript\demo-astro-react-vite\dist\
22:54:45 [build] Collecting build info...
22:54:45 [build] ✓ Completed in 281ms.
22:54:45 [build] Building static entrypoints...
22:54:46 [vite] ✓ built in 913ms
22:54:46 [build] ✓ Completed in 952ms.

 building client (vite) 
22:54:47 [vite] ✓ 74 modules transformed.
22:54:47 [vite] dist/_astro/primeicons.C6QP2o4f.woff2   35.15 kB
22:54:47 [vite] dist/_astro/primeicons.MpK4pl85.ttf     84.98 kB
22:54:47 [vite] dist/_astro/primeicons.WjwUDZjB.woff    85.06 kB
22:54:47 [vite] dist/_astro/primeicons.DMOk5skT.eot     85.16 kB
22:54:47 [vite] dist/_astro/primeicons.Dr5RGzOO.svg    342.53 kB │ gzip: 105.26 kB
22:54:47 [vite] dist/_astro/index.BeFKMXcz.css         193.09 kB │ gzip:  22.30 kB
22:54:47 [vite] dist/_astro/client.BZfCgiig.js           1.56 kB │ gzip:   0.81 kB
22:54:47 [vite] dist/_astro/DialogDemo.CqeHn6I6.js     131.82 kB │ gzip:  35.38 kB
22:54:47 [vite] dist/_astro/index.BG5Ro7KF.js          140.91 kB │ gzip:  45.31 kB
22:54:47 [vite] ✓ built in 1.02s

 generating static routes 
Named export 'Button' not found. The requested module 'primereact/button/button.esm.js' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from 'primereact/button/button.esm.js';
const { Button } = pkg;

  Stack trace:
    at ModuleJob._instantiate (node:internal/modules/esm/module_job:134:21)
    at async ModuleLoader.import (node:internal/modules/esm/loader:323:24)
    at async generatePages (file:///C:/dev/Demos/javascript/demo-astro-react-vite/node_modules/astro/dist/core/build/generate.js:164:9)
    at async AstroBuilder.build (file:///C:/dev/Demos/javascript/demo-astro-react-vite/node_modules/astro/dist/core/build/index.js:134:5)
    at async build (file:///C:/dev/Demos/javascript/demo-astro-react-vite/node_modules/astro/dist/core/build/index.js:46:3)
    
Thanks for the help.

**Comment by melloware** [comment, marked as answer]
See: https://github.com/primefaces/primereact/issues/4675
